### PR TITLE
#11544 - Error received at merge duplicate case

### LINKED ISSUE #11544: Error received at merge duplicate case
### Problem Description
Error received at merge duplicate case

#### Steps to Reproduce

1. As NatUser+Admin go to Case directory
2. Click the More button
3. Select Merge duplicates
4. Confirm the pop-up
5. Confirm the filters

### Actual Behavior
An error message appears and no list of cases returned

#### Screenshots

Screenshot 2023-02-23 at 15 16 33 (2)

#### System Details

* Device:
* SORMAS version: 1.81.0-SNAPSHOT (e64c6fd)
* Android version/Browser:
* Server URL: test-de2, test-de
* User Role:

### Additional Information

[MergeCaseError.txt](https://github.com/hzi-braunschweig/SORMAS-Project/files/10814402/MergeCaseError.txt)

### PR DESCRIPTION
**Pull request by sergiupacurariu**
Fixes #11544

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=11562

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=11562

**Review by MateStrysewske — CHANGES_REQUESTED** [review]
Had a discussion with @MartinWahnschaffe and we agreed that the solution here is kinda problematic because `AbstractUuidDto` should not have any additional fields.

Our proposed solution would be to instead override the equals and hashCode methods in CaseMergeIndexDto to work like Object.equals and Object.hashCode (i.e. only compare whether the two objects are the same and ignore the UUID); that should hopefully have the same effect.

Also, please add a class comment to CaseMergeIndexDto which explains that and why these methods need to be overridden.

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=11562

File: sormas-api/src/main/java/de/symeda/sormas/api/uuid/AbstractUuidDto.java [hunk 1]
```diff
@@ -78,11 +78,15 @@ public boolean equals(Object o) {
...
-		return uuid.equals(that.uuid);
+		return getUniqueUuid().equals(that.getUniqueUuid());
 	}
 
 	@Override
 	public int hashCode() {
 		return Objects.hash(uuid);
 	}
+
+	protected String getUniqueUuid() {
```
Comment (MateStrysewske) [inline sormas-api/src/main/java/de/symeda/sormas/api/uuid/AbstractUuidDto.java:89]: Remove this field

**Review by MateStrysewske — CHANGES_REQUESTED** [review]

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=11562

File: sormas-api/src/main/java/de/symeda/sormas/api/uuid/AbstractUuidDto.java (hunk 1 above)
Comment (sergiupacurariu) [reply sormas-api/src/main/java/de/symeda/sormas/api/uuid/AbstractUuidDto.java]: removed

**Review by MateStrysewske — APPROVED** [review]

**Pull request merged**